transmogrifier/config.py

```diff
@@ -83,10 +87,43 @@ def get_transformer(source: str) -> type:
     return getattr(source_module, class_name)
 
 
-def load_external_config(path: str) -> dict:
+def load_external_config(
+    file_path: str, file_type: Literal["json", "xml"]
+) -> Union[dict, Tag]:
+    """
+    Return dict from a JSON or XML config file. Logs a warning if a different file_type
+    parameter is passed.
+
+    """
+    with open(file_path, "rb") as config_file:
+        if file_type == "json":
+            return json.load(config_file)
+        elif file_type == "xml":
+            return BeautifulSoup(config_file, "xml")
+        else:
+            logger.error("Unrecognized file_type parameter: %s", file_type)
+            return {}
+
+
+def create_dict_from_loc_xml_config(
+    xml: Tag, element_tag: str, code_tag: str, label_tag: str
+) -> dict:
     """
-    Return dict from JSON config file.
+    Return dict from XML object that is formatted like the country and language code
+    XML files hosted by Library of Congress. It captures all authorized and obsolete
+    codes in a file.
 
+     Args:
+            xml: A BeautifulSoup Tag representing an XML config file.
+            element_tag: The element representing an item in the crosswalk, e.g.
+            language, country.
+            code_tag: The element containing a code for the item.
+            label_tag: The element containing the preferred label for the item.
     """
-    with open(path, "rb") as json_file:
-        return json.load(json_file)
+    config_dict = {}
+    for code_element in xml.find_all(code_tag):
+        for parent in [
+            parent for parent in code_element.parents if parent.name == element_tag
+        ]:
+            config_dict[str(code_element.string)] = str(parent.find(label_tag).string)
```

Comment:
We aren't testing that this code does what's expected. Also, let's think through what we want to do with obsolete codes and why. What does it mean if we encounter an obsolete code in our records? What do we want to do with that information and how do we want to handle it in the TIMDEX record?